support/src/indirect_intrusive_heap.h

```diff
@@ -196,10 +198,10 @@ namespace crimson {
     IndIntruHeap() :
       count(0)
     {
-      // empty
+      assert(K > 0);
```

Comment:
If it is left as a template argument, then I think we should see if a static (compile-time) assertion could work. If we end up making it a runtime parameter, then an assert would be fine.

Also, why not K > 1? If K is 1, then we're keeping the structure fully sorted using something like an incremental bubble sort, which isn't very efficient.